internal/versions/versions.go

```diff
@@ -24,7 +28,23 @@ func (c *baseClient) GetEngineVersions(ctx context.Context, registryClient regis
 		return clitypes.AvailableVersions{}, err
 	}
 
-	return parseTags(tags, currentVersion)
+	return parseTags(tags, serverVersion.Version)
+}
+
+func getEngineImage(registryPrefix string, serverVersion types.Version) string {
+	communityImage := "engine-community"
+	enterpriseImage := "engine-enterprise"
+	platform := strings.ToLower(serverVersion.Platform.Name)
+	if platform != "" {
+		if strings.Contains(platform, "enterprise") {
+			return path.Join(registryPrefix, enterpriseImage)
+		}
+		return path.Join(registryPrefix, communityImage)
+	}
+	if strings.Contains(serverVersion.Version, "ee") {
```

Comment:
The "ee" string lookup could theoretically be removed from this flow too, but since we're in a transitional state on the packaging some older tp builds still have ee version strings and didn't have the platform populated, so I was trying to harden this specific flow.

I'll put a comment in the code that this can be removed once we no longer have to worry about the earlier 18.09 builds that had the packaging bugs.